.github/workflows/test_blas.yml

```
@@ -35,49 +35,24 @@ jobs:
     env:
       VENV_DIR: ${{ github.workspace }}/.venv
       ARTIFACT_RUN_ID: "${{ inputs.artifact_run_id != '' && inputs.artifact_run_id || github.run_id }}"
-      BUILD_ARTIFACTS_DIR: "./build/artifacts"
-      THEROCK_BIN_DIR: "./build/artifacts/output_dir/bin"
+      OUTPUT_ARTIFACTS_DIR: "${{ github.workspace }}/build/artifacts"
+      THEROCK_BIN_DIR: "${{ github.workspace }}/build/artifacts/output_dir/bin"
 
     steps:
       - name: Checkout Repository
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
-      - name: Install the AWS tool
-        run: ./dockerfiles/cpubuilder/install_awscli.sh
-
-      - name: "Setting up Python"
-        id: setup_python
-        uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+      - name: Run setup test environment workflow
+        uses: './.github/actions/setup_test_environ'
         with:
-          python-version: 3.11
-
-      - name: Create Python venv and installing packages
-        run: |
-          python -m venv ${VENV_DIR}
-          source ${VENV_DIR}/bin/activate
-          pip install -r requirements-test.txt
-          sudo apt install gfortran -y
-
-      - name: Download and Unpack Artifacts
-        run: |
-          source ${VENV_DIR}/bin/activate
-
-          # TODO(sosa): We should infer this from the runner. For Libs and Above it should be based on the
-          # runner, everything else is generic. Use just generic since this is all core/below.
-          echo "Making ${BUILD_ARTIFACTS_DIR} directory"
-          mkdir -p "${BUILD_ARTIFACTS_DIR}"
-
-          echo "Downloading artifacts"
-          python ./build_tools/fetch_artifacts.py \
-            --run-id=${ARTIFACT_RUN_ID} \
-            --target=${{ inputs.amdgpu_families }} \
-            --build-dir=${BUILD_ARTIFACTS_DIR} \
-            --test \
-            --blas
+          ARTIFACT_RUN_ID: ${{ env.ARTIFACT_RUN_ID }}
+          AMDGPU_FAMILIES: ${{ inputs.amdgpu_families }}
+          OUTPUT_ARTIFACTS_DIR: ${{ env.OUTPUT_ARTIFACTS_DIR }}
+          VENV_DIR: ${{ env.VENV_DIR }}
+          FETCH_ARTIFACT_ARGS: "--test --blas"
 
-          echo "Unpacking artifacts"
-          mkdir ${BUILD_ARTIFACTS_DIR}/output_dir
-          python ./build_tools/fileset_tool.py artifact-flatten ${BUILD_ARTIFACTS_DIR}/*.tar.xz -o ${BUILD_ARTIFACTS_DIR}/output_dir --verbose
+      - name: Install additional packages
+        run: sudo apt install gfortran -y
```

`libgfortran5` should be sufficient. You can check with `ldd` which library is needed and `Search the contents of packages` at https://packages.ubuntu.com/ helps to identify which package provides it.

```suggestion
        run: sudo apt install libgfortran5 -y
```